minecraft/auth/xbox.go

```diff
@@ -40,8 +42,64 @@ func (t XBLToken) SetAuthHeader(r *http.Request) {
 	r.Header.Set("Authorization", fmt.Sprintf("XBL3.0 x=%v;%v", t.AuthorizationToken.DisplayClaims.UserInfo[0].UserHash, t.AuthorizationToken.Token))
 }
 
-// RequestXBLToken requests an XBOX Live auth token using the passed Live token pair.
+type XblTokenObtainer struct {
+	key         *ecdsa.PrivateKey
+	liveToken   *oauth2.Token
+	deviceToken *deviceToken
+	deviceType  *Device
+}
+
+func NewXblTokenObtainer(liveToken *oauth2.Token, deviceType *Device, ctx context.Context) (*XblTokenObtainer, error) {
```

Comment:
`NewXBLTokenObtainer`

Also would remove the pointer from `*Device`. Won't keep repeating from now on. Same counts for other functions.